### ISSUE TITLE
initial route

### ISSUE THREAD

**Issue by cyberman-io** [expo/router#428]
### Summary

i expect the initial route will be ***/chat/index*** but the initial route always ***/index*** in the route directory
File system:
- App
  - _layout
  - chat
    - [id].tsx
    - index.tsx
  - translate.tsx
  - history.tsx
  - index.tsx

### Minimal reproducible example

```
export default function Layout(){

return (
 <Drawer initialRouteName="chat">
        <Drawer.Screen
          name="chat"
          options={{
            drawerLabel: "Chat",
            title: "Chat"}}
        />
        <Drawer.Screen
          name="translate"
          options={{
            drawerLabel: "Translate",
            title: "Translate"}}
        />
        <Drawer.Screen
          name="history"
          options={{
            drawerLabel: "History",
            title: "History"}}
        />
      </Drawer>
)
}

export const unstable_settings = {
  
  initialRouteName: "chat",
};
```

**Comment by EvanBacon** [comment]
initial route name indicates which file (out of a stack) should always be rendered. If the app opens without a deep link, then the route will be `/` which matches `/index` and not `/chat`. You can setup a redirect to move pages automatically if you want.

**Comment by lucacri** [comment]
I just wasted 6 hours today trying to do the same thing you wanted/thought you could do with the "initialRouteName" thing.

Eventually i found a way to do the redirect:

```tsx
import { Redirect } from "expo-router";

const Index = () => {
	return <Redirect href="/discussions" />;
};
export default Index;
```

**Comment by Savinvadim1312** [comment]
> I just wasted 6 hours today trying to do the same thing you wanted/thought you could do with the "initialRouteName" thing.
> 
> Eventually i found a way to do the redirect:
> 
> ```tsx
> import { Redirect } from "expo-router";
> 
> const Index = () => {
> 	return <Redirect href="/discussions" />;
> };
> export default Index;
> ```

Thanks! 
@EvanBacon consider adding this to the documentation. It's a very common use-case to have some nested (tabs, drawer) screen as the default home screen. 

The redirect is not that straightforward. I spent a good amount of time trying to fix this with the `initialRouteName` and couldn't figure it out

**Comment by rudgal** [comment]
test

**Comment by rudgal** [comment]
Just used expo-router v2 for the first time and also just spent hours getting the initial route to work.

So I think this is definitely worth adding to documentation, since it is a pretty common use case as @Savinvadim1312 pointed out. Also knowing `react-navigation` one would expect the property `initialRouteName` to have an effect.

Utilizing the redirect seems to be a valid workaround, but in my case leads to the problem described in #740 .

**Comment by aadilmallick** [comment]
For issue #740 , the fix is to wait for the navigation state to be ready. This is a preliminary fix for now, but I'm hoping the documentation will get better in the coming years: 

```
import { View, Text } from "react-native";
import React from "react";
import { Redirect } from "expo-router";
import { useNavigation, useRootNavigation } from "expo-router";
import { LoadingScreen } from "../components/StyleComponents";

// index.tsx, referring to route /

export default function index() {
  const navigation = useRootNavigation();
  const [ready, setReady] = React.useState(false);

  React.useEffect(() => {
    if (!navigation?.isReady) return;

    setReady(true);
  }, [navigation?.isReady]);

  if (ready) return <Redirect href="/whatever-route" />;

  return <LoadingScreen />;
}
```

**Comment by ccctask** [comment]
it's work for me  both  some nested screens

this is my   file tree 

- app layout  is Slot 
- Drawer layout  incloud a Stack layout  inside

- [app]
  - _layout.jsx
  - [drawer]
     - _layout.jsx
     - pageOne.jsx
     - pageTwo.jsx
     - [stack]
        - _layout.jsx
        -  index.jsx
        - stackOne.jsx

> /app/_layout.jsx
```javascript
import { Slot } from "expo-router";
export default function RootLayout() {
	return <Slot />;
}
```
> /app/drawer/_layout.jsx
> init router is  /app/drawer/stack/index.jsx, its also worker when i  change  initialRouteName to  "pageOne"  the default page will be  /app/drawer/pageOne.jsx
```javascript
import { Drawer } from "expo-router/drawer";

export default function DrawerLayout() {
	return (
		<Drawer
			screenOptions={{
				drawerType: "slide",
				overlayColor: "transparent",
				drawerStyle: {
					flex: 1,
					width: "65%",
					padding: 20,
					backgroundColor: "transparent",
				},
				sceneContainerStyle: { backgroundColor: "transparent" },
			}}
			initialRouteName="stack"
		></Drawer>
	);
}
```
> /app/drawer/stack/_layout.jsx
```javascript
import { Stack } from "expo-router";

export default function Layout() {
	return (
		<Stack
			screenOptions={{
				headerShown: false,
			}}
		/>
	);
}

```

> other pages  is default template 
```javascripts
import { View, Text } from "react-native";
import React from "react";

const pageOne = () => {
	return (
		<View>
			<Text>pageOne</Text>
		</View>
	);
};

export default pageOne;
```

this is my  packages version 
`    "@react-navigation/drawer": "^6.6.3",
    "@react-navigation/native": "^6.0.2",
    "expo": "~49.0.8",
    "expo-font": "~11.4.0",
    "expo-linking": "~5.0.2",
    "expo-router": "2.0.0",`

**Comment by tonven** [comment]
@lucacri @Savinvadim1312 maybe an unrelated question. But when having this redirect in initial component, for milliseconds I can see this initial screen of index.tsx. How would you implement it, so a user will not notice this redirect?

**Comment by bnussman** [comment]
Same issue here @tonven. 

Everyone keeps suggesting redirecting, but it causes a super ugly flash of the initial screen. Is there any **real** way to handle initial routes **without** a redirect? The Expo Router docs don't provide a solution to this or even mention anything about it.


https://github.com/expo/router/assets/6440455/bb0863de-4c58-4db8-a4d3-54d00dd42e0d

**Comment by omerkaz** [comment]
> @lucacri @Savinvadim1312 maybe an unrelated question. But when having this redirect in initial component, for milliseconds I can see this initial screen of index.tsx. How would you implement it, so a user will not notice this redirect?

this is also a sad part.

**Comment by johnhaup** [comment]
@MoSattler I agree that there should be a more straightforward way to define the initial route using the router.  But I was still able to achieve what I needed using the splash screen module.  Just show your splash or some placeholder loading screen until you determine where you'd like to send the user.

https://docs.expo.dev/router/appearance/#splash-screen

**Comment by matthewhausman** [comment]
> initial route name indicates which file (out of a stack) should always be rendered. If the app opens without a deep link, then the route will be `/` which matches `/index` and not `/chat`. You can setup a redirect to move pages automatically if you want.

What if there is no index.tsx?

**Comment by bnussman** [comment]
Turns out I needed to use [Static Options](https://expo.github.io/router/docs/migration/react-navigation/screen/#static-options) to reduce flickering. [Dyanmic Options](https://expo.github.io/router/docs/migration/react-navigation/screen/#dynamic-options) inherently will cause flickering.

**Comment by filippobrigati** [comment]
Hi everyone, I'm having a problem with expo router that I can't resolve, this is my file structure

- app/
    - (tabs)
      - _layout.tsx
      - expert.tsx
      - index.tsx
      - metrics.tsx
      - settings.tsx
    - activity/
      feelings.tsx
      notes.tsx
- _layout.tsx
- modal.tsx
- sign-in.tsx
- sign-up.tsx

My goal would be to redirect the user to the page (tabs)/index by default or after logging in, and this works correcty in development with expo-go app, but when i build the app for play store i can't be redirected to index page by default, in fact, after logging in, the splash screen appears and the redirect doesn't happen...

I tried to do a different redirect after login (activity/feelings page) and it works correctly, the problem is that i cannot reach the index page.

I'm using SDK 50 and expo router 3.4.8, this in my sign-in page (only redirect method):

```
    const signInWithPassword = async () => {
      setIsLoading(true);
  
      const { data, error } = await supabase.auth.signInWithPassword({
        email: username,
        password: password,
      });
  
      if (error) {
        Alert.alert(error.message);
        setIsLoading(false);
      } else {
        await signIn(data.session.access_token, data.session.refresh_token);
        setIsLoading(false);
        console.log("After login, navigating to home page");
        router.replace("/");
      }
    };
```
  
  and this is my app/_layout.tsx

```
import type { Theme } from "@react-navigation/native";
import { DarkTheme, ThemeProvider } from "@react-navigation/native";
import { Stack } from "expo-router";
import { SafeAreaProvider } from "react-native-safe-area-context";
import { AuthProvider } from "../context/AuthContext";
import { GestureHandlerRootView } from "react-native-gesture-handler";
import * as SplashScreen from "expo-splash-screen";
import { useColorScheme } from "react-native";

export const unstable_settings = {
  initialRouteName: "(tabs)",
};

// Prevent the splash screen from auto-hiding before asset loading is complete.
SplashScreen.preventAutoHideAsync();

export default function Root() {
  const colorScheme = useColorScheme();

  const DefaultTheme: Theme = {
    dark: false,
    colors: {
      primary: "rgb(0, 122, 255)",
      background: "rgb(255, 255, 255)",
      card: "rgb(255, 255, 255)",
      text: "rgb(28, 28, 30)",
      border: "rgb(216, 216, 216)",
      notification: "rgb(255, 59, 48)",
    },
  };

  SplashScreen.hideAsync();

  return (
    <ThemeProvider value={colorScheme === "dark" ? DarkTheme : DefaultTheme}>
      <AuthProvider>
        <GestureHandlerRootView style={{ flex: 1 }}>
          <SafeAreaProvider>
            <Stack>
              <Stack.Screen name="(tabs)" options={{ headerShown: false }} />
              <Stack.Screen name="sign-in" options={{ headerShown: false }} />
              <Stack.Screen name="sign-up" options={{ headerShown: false }} />
              <Stack.Screen
                name="activity/note"
                options={{ headerShown: false }}
              />
              <Stack.Screen
                name="activity/feelings"
                options={{ headerShown: false }}
              />
              <Stack.Screen name="modal" options={{ presentation: "modal" }} />
            </Stack>
          </SafeAreaProvider>
        </GestureHandlerRootView>
      </AuthProvider>
    </ThemeProvider>
  );
}
```
and finally this is my app/(tabs)/layout.tsx

```import { Redirect, Tabs } from "expo-router";
import { useAuth } from "../../context/AuthContext";
import { useColorScheme, Text } from "react-native";

import { SimpleLineIcons } from "@expo/vector-icons";
import { AntDesign } from "@expo/vector-icons";

export default function TabLayout() {
  const { isLoading, userToken } = useAuth();
  const colorScheme = useColorScheme();
  // You can keep the splash screen open, or render a loading screen like we do here.
  if (isLoading) {
    return <Text>Loading...</Text>;
  }

  // need to be able to access the (auth) group and sign in again.
  if (!userToken) {
    // On web, static rendering will stop here as the user is not authenticated
    // in the headless Node process that the pages are rendered in.
    return <Redirect href="/sign-in" />;
  }

  return (
    <Tabs
      screenOptions={{
        tabBarActiveTintColor: "#2f95dc",
        headerShown: false,
        tabBarShowLabel: false,
      }}
    >
      <Tabs.Screen
        name="index"
        options={{
          title: "Home",
          tabBarIcon(props) {
            return <AntDesign name="home" size={24} color={colorScheme === "light" ? (props.focused ? "black" : "gray") : (props.focused ? "white" : "#ced4da")} />
          }
        }}
      />
      <Tabs.Screen
        name="metrics"
        options={{
          title: "Metrics",
          tabBarIcon(props) {
            return <SimpleLineIcons name="graph" size={24} color={colorScheme === "light" ? (props.focused ? "black" : "gray") : (props.focused ? "white" : "#ced4da")} />
          }
        }}
      />
      <Tabs.Screen
        name="expert"
        options={{
          title: "Expert",
          tabBarIcon(props) {
            return <AntDesign name="staro" size={24} color={colorScheme === "light" ? (props.focused ? "black" : "gray") : (props.focused ? "white" : "#ced4da")} />
          }
        }}
      />
      <Tabs.Screen
        name="settings"
        options={{
          title: "Impostazioni",
          tabBarIcon(props) {
            return <AntDesign name="setting" size={24} color={colorScheme === "light" ? (props.focused ? "black" : "gray") : (props.focused ? "white" : "#ced4da")} />
          },
        }}
      />
    </Tabs>
  );
}
```

@EvanBacon I'm probably doing something wrong, but I can't figure out where.

Thanks everyone for the help!

**Comment by ArthurDias01** [comment]
I spent literally more than 2 days in this, and still can't figure it out, seems like this is very inconsistent as expo-router v3

```ts

import { queryClient } from "@/src/lib/query";
import { useReactQueryDevTools } from "@dev-plugins/react-query";
import FontAwesome from "@expo/vector-icons/FontAwesome";
import { QueryClientProvider } from "@tanstack/react-query";
import { useFonts } from "expo-font";
import * as SplashScreen from "expo-splash-screen";
import { useEffect } from "react";
import "react-native-gesture-handler";
import { ThemeProvider } from "styled-components";

import {
  Inter_400Regular,
  Inter_500Medium,
  Inter_700Bold,
} from "@expo-google-fonts/inter";

import {
  Nunito_400Regular,
  Nunito_600SemiBold,
  Nunito_700Bold,
} from "@expo-google-fonts/nunito";

import theme from "@/global/styles/theme";
import { AuthProvider } from "@/src/providers/auth";
import { ConfigNotificationContextProvider } from "@/src/providers/config-notifications-ctx";
import { UpdateProvider } from "@/src/providers/update-ctx";
import { DrawerToggleButton } from "@react-navigation/drawer";
import Drawer from "expo-router/drawer";
import { GestureHandlerRootView } from "react-native-gesture-handler";

export {
  // Catch any errors thrown by the Layout component.
  ErrorBoundary
} from "expo-router";

export const unstable_settings = {
  // Ensure that reloading on `/modal` keeps a back button present.
  initialRouteName: "(tabs)",
};

// Prevent the splash screen from auto-hiding before asset loading is complete.
SplashScreen.preventAutoHideAsync();

export default function RootLayout() {
  useReactQueryDevTools(queryClient);
  const [loaded, error] = useFonts({
    SpaceMono: require("@assets/fonts/SpaceMono-Regular.ttf"),
    Inter_400Regular,
    Inter_500Medium,
    Inter_700Bold,
    Nunito_400Regular,
    Nunito_600SemiBold,
    Nunito_700Bold,
    ...FontAwesome.font,
  });

  // Expo Router uses Error Boundaries to catch errors in the navigation tree.
  useEffect(() => {
    if (error) throw error;
  }, [error]);

  // Hide the splash screen once the app is ready. (fonts loaded)
  useEffect(() => {
    if (loaded) {
      SplashScreen.hideAsync();
    }
  }, [loaded]);

  if (!loaded) {
    return null;
  }

  return <RootLayoutNav />;
}

function RootLayoutNav() {
  return (
    <ThemeProvider theme={theme}>
      <QueryClientProvider client={queryClient}>
        <UpdateProvider>
          <AuthProvider>
            <ConfigNotificationContextProvider>
              <GestureHandlerRootView style={{ flex: 1 }}>
                <Drawer
                  screenOptions={{
                    headerShown: false,
                    drawerActiveTintColor: theme.colors.primary,
                    drawerInactiveTintColor: theme.colors.text,
                    swipeEdgeWidth: 80,
                  }}
                  initialRouteName="/"
                >
                  <Drawer.Screen
                    name="index"
                    options={{
                      drawerLabel: "Tabs",
                      headerLeft: () => <DrawerToggleButton />,
                      drawerItemStyle: { display: "none" },
                    }}
                  />
                  <Drawer.Screen
                    name="(tabs)"
                    options={{
                      drawerLabel: "Tabs",
                      headerLeft: () => <DrawerToggleButton />,
                    }}
                  />
                  <Drawer.Screen
                    name="minha-conta"
                    options={{
                      drawerLabel: "Minha Conta",
                      headerLeft: () => <DrawerToggleButton />,
                    }}
                  />
                  <Drawer.Screen
                    name="anunciar"
                    options={{
                      drawerLabel: "anunciar",
                      headerLeft: () => <DrawerToggleButton />,
                    }}
                  />
                  <Drawer.Screen
                    name="+not-found"
                    options={{
                      drawerLabel: "anunciar",
                      headerLeft: () => <DrawerToggleButton />,
                      drawerItemStyle: { display: "none" },
                    }}
                  />
                </Drawer>
              </GestureHandlerRootView>
            </ConfigNotificationContextProvider>
          </AuthProvider>
        </UpdateProvider>
      </QueryClientProvider>
    </ThemeProvider>
  );
}
```

even creating the index with redirect will not redirect to the correct route.

![image](https://github.com/expo/router/assets/83284629/0b86f75a-8f52-4d78-9b6a-d57e7951190c)

https://github.com/expo/router/assets/83284629/6332b5cf-29b5-47b9-8c34-95dde7d3226c

**Comment by marklawlor** [comment]
@ArthurDias01 This repo is for v2 and is only maintained for security purposes. If you are having an issue, please create a new issue at https://github.com/expo/expo

Also, reporting an issue with a code snippet of a single file **does not** met the criteria for a minimal reproduce-able repo. I can see from your video that the app is crashing from an error - but you have not provided me enough information to advise on what that error is. The best method is simply to share your repo so I can run the project.

**Comment by ArthurDias01** [comment]
I got it to work after some heavy debugging,  clearing all cache, reinstalling all dependencies and creating a new clean build. @marklawlor thanks. Will try to be more specific to the issue next time.

**Comment by alexmlong** [comment]
Hello, why was this issue closed? I'm having this issue also. And the empty "index" also appears as an extra tab in the tab bar. Would love to get a solution to this! 🙏

**Comment by a-eid** [comment]
> I just wasted 6 hours today trying to do the same thing you wanted/thought you could do with the "initialRouteName" thing.
> 
> Eventually i found a way to do the redirect:
> 
> ```tsx
> import { Redirect } from "expo-router";
> 
> const Index = () => {
> 	return <Redirect href="/discussions" />;
> };
> export default Index;
> ```

the only issue I see with this is the transition / animation.

**Comment by 0xbenon** [comment]
> > I just wasted 6 hours today trying to do the same thing you wanted/thought you could do with the "initialRouteName" thing.
> > Eventually i found a way to do the redirect:
> > ```tsx
> > import { Redirect } from "expo-router";
> > 
> > const Index = () => {
> > 	return <Redirect href="/discussions" />;
> > };
> > export default Index;
> > ```
> 
> the only issue I see with this is the transition / animation.

Yeah, the user can see a white screen before redirect

**Comment by Diego-Fdez** [comment]
> Acabo de perder 6 horas hoy tratando de hacer lo mismo que querías / pensabas que podías hacer con lo de "initialRouteName".
> 
> Finalmente encontré una manera de hacer la redirección:
> 
> ```tsx
> import { Redirect } from "expo-router";
> 
> const Index = () => {
> 	return <Redirect href="/discussions" />;
> };
> export default Index;
> ```

This worked for me...

**Comment by fredrikburmester** [comment]
Is there no fix to this?

**Comment by KovalevAnton** [comment]
It seems like I’m not the only one having this problem. I’m using **< Slot />** with initialRouteName, but it doesn’t seem to have any effect. It always shows '/' instead of '/home'.

**Comment by Samox** [comment]
Same. I'm creating a stack navigator
- app
  - onboarding
    - questions
      - [id]
      
And when I navigate to /onboarding/questions I'd like to be on /onboarding/questions/0 by default to have an **id** = 0. For now I navigate to /onboarding/questions, the component `[id].tsx` is properly read but with **id** being undefined, I default it to 0 manually for now but I'm not confortable with that

**Comment by okalil** [comment]
Hey. To avoid the redirect flicker my workaround is delaying the SplashScreen hide, and optionally disable the animation for the target route:

```tsx
import { Stack, SplashScreen } from 'expo-router';

SplashScreen.preventAutoHideAsync();

export default function Layout() {
  useEffect(() => {
    requestAnimationFrame(SplashScreen.hideAsync);
  }, []);
  return (
    <Stack screenOptions={{ headerShown: false }}>
      <Stack.Screen name="index" />
      <Stack.Screen name="[user]" options={{ animation: 'none' }} />
    </Stack>
  );
}
```

But I noticed that looks like there is no flicker when using redirect attribute instead of using `<Redirect />` inside screen component:
```tsx
import { Stack } from 'expo-router';

export default function Layout() {
  return (
    <Stack screenOptions={{ headerShown: false }}>
      <Stack.Screen name="index" redirect />
      <Stack.Screen name="[user]" />
    </Stack>
  );
}
```

**Comment by sanny-io** [comment]
> initial route name indicates which file (out of a stack) should always be rendered. If the app opens without a deep link, then the route will be `/` which matches `/index` and not `/chat`. You can setup a redirect to move pages automatically if you want.

That's pretty bizarre to hear. It literally does not do what it says it does lol.

**Comment by MackKaputo** [comment]
I had the issue after upgrading to expo 52. Solution has been downgrading expo-router to the previous version before upgrade: from version  ~4.0.6 to  3.5.24

**Comment by kot-ezhva** [comment]
Working example with selected any tabbar and hidden index-tabBaritem:
_layout.txs
```
return (
    <Tabs
      // tabBar={(props) => <TabBar {...props} />}
      screenOptions={{ headerShown: false }}
    >
      <Tabs.Screen name="(history)" />
      <Tabs.Screen name="(scan)" />
      <Tabs.Screen name="(settings)" />


      {/** This shit needs for select scan-stack as initial route, see index.tsx **/}
      <Tabs.Screen name="index" options={{ href: null }} />
    </Tabs>
  );
```
index.tsx inside tabs directory
```
import React from 'react';
import { Redirect } from 'expo-router';

const Index = () => <Redirect href="/(scan)" />;

export default Index;
```

Tested on last Expo SDK("expo": "~52.0.11",) and "expo-router": "~4.0.9", "@react-navigation/native": "^7.0.0",

**Comment by georrgee** [comment]
> Hey. To avoid the redirect flicker my workaround is delaying the SplashScreen hide, and optionally disable the animation for the target route:
> 
> ```tsx
> import { Stack, SplashScreen } from 'expo-router';
> 
> SplashScreen.preventAutoHideAsync();
> 
> export default function Layout() {
>   useEffect(() => {
>     requestAnimationFrame(SplashScreen.hideAsync);
>   }, []);
>   return (
>     <Stack screenOptions={{ headerShown: false }}>
>       <Stack.Screen name="index" />
>       <Stack.Screen name="[user]" options={{ animation: 'none' }} />
>     </Stack>
>   );
> }
> ```
> 
> But I noticed that looks like there is no flicker when using redirect attribute instead of using `<Redirect />` inside screen component:
> 
> ```tsx
> import { Stack } from 'expo-router';
> 
> export default function Layout() {
>   return (
>     <Stack screenOptions={{ headerShown: false }}>
>       <Stack.Screen name="index" redirect />
>       <Stack.Screen name="[user]" />
>     </Stack>
>   );
> }
> ```

This definitely worked for me. Thank you! No flicker happening

**Comment by hlspablo** [comment]
**The redirect worked to my use case too**, just to complement, here is what redirect does:
"Redirect to the nearest sibling route. 
If all children are redirect={true}, the layout will render null as there are no children to render."

So it's possible to render the next stack based on some boolean passed to redirect.

**Comment by den-ii** [comment]
For my case, it wasn't working for nested groups.  I'm giving my initial route a "()" or start it with an "a" since it work in alphabetical order.

**Comment by csherm08** [comment]
> it's work for me both some nested screens
> 
> this is my file tree
> 
> * app layout  is Slot
> * Drawer layout  incloud a Stack layout  inside
> * [app]
>   
>   * _layout.jsx
>   * [drawer]
>     
>     * _layout.jsx
>     * pageOne.jsx
>     * pageTwo.jsx
>     * [stack]
>       
>       * _layout.jsx
>       * index.jsx
>       * stackOne.jsx
> 
> > /app/_layout.jsx
> 
> ```js
> import { Slot } from "expo-router";
> export default function RootLayout() {
> 	return <Slot />;
> }
> ```
> 
> > /app/drawer/_layout.jsx
> > init router is  /app/drawer/stack/index.jsx, its also worker when i  change  initialRouteName to  "pageOne"  the default page will be  /app/drawer/pageOne.jsx
> 
> ```js
> import { Drawer } from "expo-router/drawer";
> 
> export default function DrawerLayout() {
> 	return (
> 		<Drawer
> 			screenOptions={{
> 				drawerType: "slide",
> 				overlayColor: "transparent",
> 				drawerStyle: {
> 					flex: 1,
> 					width: "65%",
> 					padding: 20,
> 					backgroundColor: "transparent",
> 				},
> 				sceneContainerStyle: { backgroundColor: "transparent" },
> 			}}
> 			initialRouteName="stack"
> 		></Drawer>
> 	);
> }
> ```
> 
> > /app/drawer/stack/_layout.jsx
> 
> ```js
> import { Stack } from "expo-router";
> 
> export default function Layout() {
> 	return (
> 		<Stack
> 			screenOptions={{
> 				headerShown: false,
> 			}}
> 		/>
> 	);
> }
> ```
> 
> > other pages  is default template
> 
> ```
> import { View, Text } from "react-native";
> import React from "react";
> 
> const pageOne = () => {
> 	return (
> 		<View>
> 			<Text>pageOne</Text>
> 		</View>
> 	);
> };
> 
> export default pageOne;
> ```
> 
> this is my packages version ` "@react-navigation/drawer": "^6.6.3", "@react-navigation/native": "^6.0.2", "expo": "~49.0.8", "expo-font": "~11.4.0", "expo-linking": "~5.0.2", "expo-router": "2.0.0",`

I need you to know this saved me after weeks

**Comment by den-ii** [comment]
This works. I was using this 

```
export const unstable_settings = {
  initialRouteName: "(home)",
};
```

but then changed to this and it works

```
export const unstable_settings = {
  initialRouteName: "(home)/index",
};
```

My code:
```
export const unstable_settings = {
  initialRouteName: "(home)/index",
};

export default function TabLayout() {
  const auth = true;
  const first_time = false;

  if (!auth) {
    if (first_time) return <Redirect href={"/(onboarding)/"} />;
    return <Redirect href={"/(auth)/"} />;
  }

  return (
    <Tabs
      screenOptions={{
        headerShown: false,
        tabBarShowLabel: true,
        tabBarActiveTintColor: Colors.light.text,
        tabBarInactiveTintColor: Colors.light.tabIconDefault,
        tabBarLabelStyle: {
          fontSize: 12,
          fontFamily: "Inter_600SemiBold",
        },
      }}
    >
      <Tabs.Screen
        name="(home)"
        options={{
          tabBarLabel: "Home",
          tabBarIcon: ({ focused }) => (
            <Home
              fill={
                focused
                  ? Colors.light.tabIconSelected
                  : Colors.light.tabIconDefault
              }
            />
          ),
        }}
      />
      <Tabs.Screen
        name="(beneficiary)"
        options={{
          tabBarLabel: "Beneficiary",
          tabBarIcon: ({ focused }) => (
            <TabBeneficiary
              fill={
                focused
                  ? Colors.light.tabIconSelected
                  : Colors.light.tabIconDefault
              }
            />
          ),
        }}
      />
      <Tabs.Screen
        name="history"
        options={{
          tabBarLabel: "History",
          tabBarIcon: ({ focused }) => (
            <TabHistory
              fill={
                focused
                  ? Colors.light.tabIconSelected
                  : Colors.light.tabIconDefault
              }
            />
          ),
        }}
      />
      <Tabs.Screen
        name="more"
        options={{
          tabBarLabel: "More",
          tabBarIcon: ({ focused }) => (
            <More
              fill={
                focused
                  ? Colors.light.tabIconSelected
                  : Colors.light.tabIconDefault
              }
            />
          ),
        }}
      />
    </Tabs>
  );
}
```

**Comment by ozturksirin** [comment]
The problem for me was the folder structure, the folder I had just created was at the top so it was getting confused, when I redirected I could not use that page again. 
I edited initialRoute to the folder I wanted it to be (00-foldername) and the problem was solved

**Comment by rocknegi** [comment]
The simplest and easiest way to fix the issue is
1. use app/index.tsx
2. folder structure for reference 
![image](https://github.com/user-attachments/assets/7915e8a2-e8c1-473a-8694-1c9b8d67a6c8) 
3.  have your app/_layout.tsx like this
```
          <Stack
              screenOptions={{
                headerShown: false,
              }}
              initialRouteName="(tabs)"
            >
              <Stack.Screen name="index" redirect />
              <Stack.Screen name="(tabs)" options={{ headerShown: false }} />
            </Stack>
```
4. app/index.tsx like this 
```
export default function index() {
  return null;
}
```
5. tabs/_layout.tsx
```
<Tabs
      screenOptions={{
        tabBarHideOnKeyboard: true,
      }}
      tabBar={(props) => <BottomNavigationBar {...props} />}
    >
      <Tabs.Screen
        name="dashboard"
        options={{
          title: "Dashboard",
          headerTitle: (props) => <Header />,
        }}
      />
      <Tabs.Screen
        name="diary"
        options={{
          title: "Diary",
        }}
      />
      <Tabs.Screen
        name="settings"
        options={{
          title: "Settings",
        }}
      />
    </Tabs>
```
6. Then your app will run like this. Enjoy!!!

https://github.com/user-attachments/assets/7e670bcc-f08e-45f8-8033-5a76e62dd62f

**Comment by russeg** [comment]
@EvanBacon you should read the comments here. 

expo router has a design implementation issue, ie file based routing is not hard to understand but your implementation is. really no better than react-navigation in the unintuitiveness factor.
